src/main/java/graphql/execution/ResultPath.java

```diff
@@ -294,15 +310,7 @@ public List<String> getKeysOnly() {
      */
     @Override
     public String toString() {
-        if (parent == null) {
-            return "";
-        }
-
-        if (ROOT_PATH.equals(parent)) {
-            return segmentToString();
-        }
-
-        return parent.toString() + segmentToString();
+        return toStringValue;
```

Comment:
Perhaps this could be done lazily?

Reply:
yes could be, but I decided that the overhead of the lazy initialization is not worth it.